hub/core/version_control/test_version_control.py

```diff
@@ -340,25 +340,127 @@ def test_auto_checkout(local_ds):
 
 def test_auto_commit(local_ds):
     initial_commit_id = local_ds.commit_id
+    # auto commit as head of main branch
     local_ds.checkout("pqr", create=True)
     local_ds.checkout("main")
-    # no auto commit as there was no data
-    assert local_ds.commit_id == initial_commit_id
+    second_commit_id = local_ds.commit_id
+    assert second_commit_id != initial_commit_id
     local_ds.create_tensor("abc")
     local_ds.abc.append(1)
-    # auto commit as there was data
+    # auto commit as head of main again
     local_ds.checkout("xyz", create=True)
     local_ds.checkout("main")
 
-    assert local_ds.commit_id != initial_commit_id
+    assert local_ds.commit_id != second_commit_id
 
     with local_ds:
         local_ds.abc.append(1)
 
-    current_commit_id = local_ds.commit_id
+    third_commit_id = local_ds.commit_id
 
-    # auto commit as there was data
+    # auto commit as head of main again
     local_ds.checkout("tuv", create=True)
     local_ds.checkout("main")
 
-    assert local_ds.commit_id != current_commit_id
+    assert local_ds.commit_id != third_commit_id
+
+
+def test_diff_linear(local_ds):
+    with local_ds:
+        local_ds.create_tensor("xyz")
+        local_ds.xyz.extend([1, 2, 3])
+        local_ds.create_tensor("pqr")
+        local_ds.pqr.extend([4, 5, 6])
+    a = local_ds.commit()
+    with local_ds:
+        local_ds.xyz[0] = 10
+        local_ds.pqr[2] = 20
+    local_ds.diff()
+    b = local_ds.commit()
+    print("blank")
+    local_ds.diff()
+    print("a")
+    local_ds.diff(a)
+    print("b")
+    local_ds.diff(b)
+    print("a,b")
+    local_ds.diff(a, b)
+    print("b,a")
+    local_ds.diff(b, a)
```

Comment:
There no asserts in the tests. Please refactor it.

Reply:
Ah yes, these were local tests I was playing around with, fill add assertions.